### PR TITLE
Fix console error

### PR DESCRIPTION
**Pull request by Latropos**
## Summary

Fix this issue: https://github.com/software-mansion/react-native-reanimated/issues/4225

Looks like we have been overwriting console and making its properties not configurable. This PR should fix the problem.

## Test plan
Tested on example app

### CODE REVIEW

File: src/reanimated2/initializers.ts [hunk 1]
```diff
@@ -166,16 +168,20 @@ export function initializeUIRuntime() {
...
+      writable: true,
     };
 
+    Object.defineProperties(global.console, {
+      debug: { ...props, value: runOnJS(capturableConsole.debug) },
+      log: { ...props, value: capturableConsole.log },
+      warn: { ...props, value: capturableConsole.warn },
+      error: { ...props, value: capturableConsole.error },
+      info: { ...props, value: capturableConsole.info },
+    });
```
[the remark this answers is not in the document]
Comment (Latropos) [reply src/reanimated2/initializers.ts:183]: We don't need it, but it lets us get rid of `@ts-ignore`

[the remark this answers is not in the document]
Comment (mlazari) [reply src/reanimated2/initializers.ts:183]: I mean someone in that issue mentioned that they commented out the global.console assignment. Should they expect that something will not work without it? I just downgraded back to v2 instead for now until this is fixed.

**Comment by freeboub** [comment]
Tested locally on my side, I confirm it fixes the issue

File: src/reanimated2/initializers.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomekzaw) [reply src/reanimated2/initializers.ts:183]: > We don't need it, but it lets us get rid of @ts-ignore

I think the problem with TypeScript is rather related to the fact that we don't implement all `console.*` methods but only a subset of them.

> why only debug is wrapped in `runOnJS` and the other ones not?

+1 to this question

[the remark this answers is not in the document]
Comment (freeboub) [reply src/reanimated2/initializers.ts:183]: Is it really expected to remove all : runOnJS except on debug ?

**Comment by piaskowyk** [comment]
@Latropos could you test it in release build also?

File: src/reanimated2/initializers.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (piaskowyk) [reply src/reanimated2/initializers.ts:183]: In the current implementation all console.* methods are wrapped by `runOnJS`, I don't think we want to remove it now. 
https://github.com/software-mansion/react-native-reanimated/blob/main/src/reanimated2/initializers.ts#L172-L176

**Comment by Latropos** [comment]
@freeboub The crash on release was because of the refactor - we can't call defineProperties on console on release build, since console in not an object on release. Hope it's fixed
> TypeError: Object.defineProperties() called on non-object

**Comment by mlazari** [comment]
I can confirm that the "property is not configurable" error does not occur with the latest change in this PR and the release build does not crash in my repro project: https://github.com/mlazari/ReanimatedRepro/tree/fix-pr

**Comment by mlazari** [comment]
It looks like the global.console assignment affects console.trace(): https://github.com/mlazari/ReanimatedRepro/blob/console-trace/App.tsx#L63

If I comment out the global.console assignment:
Screenshot 2023-03-23 at 12 53 04

With the global.console assignment:
Screenshot 2023-03-23 at 12 52 36

**Comment by piaskowyk** [comment]
@mlazari I wasn't able to reproduce this problem with `console.trace()`. Moreover, we modify just `global.console` on Reanimated JS Runtime, we don't change the console object form React Runtime.

**Review by piaskowyk — APPROVED** [review]

**Comment by chakravarthi-bb** [comment]
Can we make a release for this patch, please??

**Comment by tiagomsmagalhaes** [comment]
@chakravarthi-bb you can append `.patch` to this url to generate a patch file. just need afterwards to change the path on the diff to the `node_modules` directory of the package

**Comment by altarrok** [comment]
+1 on making a release for this patch

**Comment by monicatiba** [comment]
Do we have a fix for this? I'm facing the same error even with the last version of the lib and I cannot find a way to fix it. I tried everything


Simulator Screenshot - iPhone 14 - 2023-07-03 at 09 32 44

**Comment by mlazari** [comment]
@monicatiba Are you sure that is related to react-native-reanimated? The stack trace looks different from the one in https://github.com/software-mansion/react-native-reanimated/issues/4225 and as far as I can tell that is fixed in the latest version.

**Comment by ZakaryH** [comment]
@monicatiba I had this same error and stack trace. for me the problem was that I was applying a `useAnimatedStyle` style to a `View` rather than an `Animated.View`.

**Pull request merged**